File: cse220-datastructures/all-py-files/ll_stack.py
class Node:
    def __init__(self, data, _next) -> None:
        self.data = data
        self.next = _next


class Stack:
    def __init__(self) -> None:
        # dummy headed
        self.head = Node(None, None)

    def push(self, data):
        new_node = Node(data, self.head.next)
        self.head.next = new_node

    def peek(self):
        if self.head.next != None:
            return self.head.next.data
        else:
            print("The stack is empty")

    def pop(self):
        if self.head.next != None:
            tmp = self.head.next
            self.head.next = tmp.next
            del tmp
        else:
            print("The stackis empty")

    def __str__(self):
        output = ""
        tmp = self.head.next
        while tmp != None:
            output += f"({tmp.data}) -> "
            tmp = tmp.next
        return output[:-4]


if __name__ == "__main__":
    s = Stack()
    s.push(1)
    s.push(2000)
    s.push(30)
    s.push(4)
    s.push(5)
    s.push(60)
    print(s)
    print(s.peek())

File: cse220-datastructures/all-py-files/LLSTack.java
public class LLSTack {

    public static void main(String[] args) {
        Stack s = new Stack();
        s.push(30);
        s.push(40);
        s.push(51);
        s.print();
        s.pop();
        System.out.println(s.peek());
    }
}

class Node {
    int data;
    Node next;

    public Node(int data, Node next) {
        this.data = data;
        this.next = next;
    }
}

class Stack {
    Node head;

    public void push(int data) {
        Node newNode = new Node(data, this.head);
        this.head = newNode;
    }

    public int peek() {
        if (this.head == null) {
            System.out.println("The stack is empty!");
            // return null;
            return -1;
        } else {
            return this.head.data;
        }
    }

    public int pop() {
        if (this.head == null) {
            System.out.println("The stack is empty!");
            // return null;
            return -1;
        } else {
            Node tmp = this.head;
            this.head = tmp.next;
            return tmp.data;
        }

    }

    public void print() {
        Node tmp = this.head;
        while (tmp != null) {
            System.out.print(tmp.data + " ");
            tmp = tmp.next;
        }
        System.out.println();
    }
}
